packages/game-passes/test/GamePasses.test.ts

```diff
@@ -0,0 +1,1251 @@
+            price,
+            deadline,
+            signature,
+          ),
+      ).to.be.revertedWithCustomError(sandboxPasses, 'InvalidSigner');
+    });
+
+    it('should not allow exceeding max per wallet', async function () {
+      const price = ethers.parseEther('0.1');
+      const deadline = (await time.latest()) + 3600;
+      const nonce = 0;
+
+      // Approve payment token
+      await paymentToken
+        .connect(user1)
+        .approve(await sandboxPasses.getAddress(), price);
+
+      // Create signature for more than max per wallet
+      const signature = await createMintSignature(
+        signer,
+        user1.address,
+        TOKEN_ID_1,
+        MAX_PER_WALLET + 1,
+        price,
+        deadline,
+        nonce,
+      );
+
+      // Try to mint more than max per wallet
+      await expect(
+        sandboxPasses
+          .connect(user1)
+          .mint(
+            user1.address,
+            TOKEN_ID_1,
+            MAX_PER_WALLET + 1,
+            price,
+            deadline,
+            signature,
+          ),
+      ).to.be.revertedWithCustomError(sandboxPasses, 'ExceedsMaxPerWallet');
+    });
+  });
+
+  describe('Burn and Mint Operations', function () {
+    beforeEach(async function () {
+      // Mint tokens to burn
+      await sandboxPasses
+        .connect(admin)
+        .adminMint(user1.address, TOKEN_ID_1, MINT_AMOUNT);
+    });
+
+    it('should allow operator to burn and mint', async function () {
+      await sandboxPasses
+        .connect(operator)
+        .operatorBurnAndMint(
+          user1.address,
+          user2.address,
+          TOKEN_ID_1,
+          2,
+          TOKEN_ID_2,
+          3,
+        );
+
+      expect(await sandboxPasses.balanceOf(user1.address, TOKEN_ID_1)).to.equal(
+        MINT_AMOUNT - 2,
+      );
+      expect(await sandboxPasses.balanceOf(user2.address, TOKEN_ID_2)).to.equal(
+        3,
+      );
+    });
+
+    it('should allow operator to batch burn and mint', async function () {
+      // Mint additional tokens for batch burn
+      await sandboxPasses
+        .connect(admin)
+        .adminMint(user1.address, TOKEN_ID_2, MINT_AMOUNT);
+
+      await sandboxPasses
+        .connect(operator)
+        .operatorBatchBurnAndMint(
+          user1.address,
+          user2.address,
+          [TOKEN_ID_1, TOKEN_ID_2],
+          [2, 3],
+          [TOKEN_ID_2, TOKEN_ID_1],
+          [4, 5],
+        );
+
+      expect(await sandboxPasses.balanceOf(user1.address, TOKEN_ID_1)).to.equal(
+        MINT_AMOUNT - 2,
+      );
+      expect(await sandboxPasses.balanceOf(user1.address, TOKEN_ID_2)).to.equal(
+        MINT_AMOUNT - 3,
+      );
+      expect(await sandboxPasses.balanceOf(user2.address, TOKEN_ID_1)).to.equal(
+        5,
+      );
+      expect(await sandboxPasses.balanceOf(user2.address, TOKEN_ID_2)).to.equal(
+        4,
+      );
+    });
+
+    it('should allow user to burn and mint with valid signature', async function () {
```

Comment:
missing bad signature tests

Reply:
Good call, added in https://github.com/thesandboxgame/sandbox-smart-contracts/pull/1619/commits/e515992d2fe7fdf5a993405c21d386b7f11938ce